### DISCUSSION TITLE
Filter on empty array in the js client.

### DISCUSSION THREAD

**Discussion by nilsi** [supabase/supabase#19754] [Questions] [answered]
Im trying to do the following

```
export async function getSearchListings(searchParams: ReadonlyURLSearchParams): Promise<ListingData[]> {
  const supabase = createClient();

  const params = new URLSearchParams(searchParams);
  const searchTerm = params.get("location")
  const facilities = params.get("facilities")
  const startDate = params.get("checkIn")
  const endDate = params.get("checkOut")

  const query = supabase.from('listings')
    .select(`
      id,
      availability(
        *
      )
    `)
    .eq('published', true)

  if (facilities) {
    query.contains('facilities', JSON.stringify(facilities.split(',')))
  }

  if (searchTerm) {
    query.textSearch('county_location', searchTerm, {
      type: 'websearch',
      config: 'swedish',
    })
  }

  if (startDate) {
    query.lte('availability.start', startDate)
  }

  if (endDate) {
    query.gte('availability.end', endDate)
  }

  const { data, error } = await query


  if (error) {
    console.log(error.message);
    return [];
  }

  return data || [];
}
```

This works great. Now I want to remove the results where the referenced table availability is empty. 
I tried with adding a filter like  `query.gt('availability.length', 0)` but this does not seem to work.

Is there a way I can filter out the results where availability is empty?

Thank you!

**Comment by GaryAustin1** [comment, marked as answer]
If you are simply saying if there is not a relationship found to meet the filter don't include the parent row try adding !inner like 
```
select(`
      id,
      availability!inner(
        *
      )
    `)
    ```
    See the notes here:  https://supabase.com/docs/reference/javascript/select?example=filtering-through-referenced-tables

**Reply by nilsi** [reply]
Oh, wow! that seems to do the trick. I don't fully get what it does but big thanks for the help!

**Reply by GaryAustin1** [reply]
Basically it is telling the API to do an SQL inner join.  This means only rows in the parent table that match the filter for the embedded(foreign) table are included.

**Reply by nishantbhavasar** [reply]
@GaryAustin1 Hello,
i have similer problem in join's but in my case i want those column's which don't have data means 
in this answer this query return those column which have data but i want those column which not contain data like empty column's is this possible or yes please please provide me answer
